packages/react/src/components/List/ListContent/ListContent.jsx

```diff
@@ -257,10 +262,30 @@ const ListContent = ({
       ...(!hasChildren && item.hasLoadMore ? [renderLoadMore(item, isLoadingMore, level)] : []),
     ];
   };
+  const previousExpandedIds = usePrevious(expandedIds, expandedIds);
+  // get the lastId of the diff between previous and current, this handles closes
+  const [previousLastId] = previousExpandedIds.filter((id) => !expandedIds.includes(id)).slice(-1);
```

Comment:
Sorry, just realised that we can perhaps do a little more for performance here. We can't use `useMemo` but I assume lodash `isEqual` is still more efficient than comparing every item in X with every item in Y.

```
const selectionChanged = !isEqual(expandedIds, previousExpandedIds);
const previousLastId = selectionChanged && previousExpandedIds.filter((id) => !expandedIds.includes(id)).slice(-1)[0];
...
```

it might require some additional changes to get this to work

Reply:
Done.

Reply:
Sweet, I tested it, it works fine and you already have the approval.